adapters/missena/missena.go

```diff
@@ -0,0 +1,226 @@
+	missenaRequest := MissenaAdRequest{
+		RequestId:        request.ID,
+		Timeout:          2000,
+		Referer:          request.Site.Page,
+		RefererCanonical: request.Site.Domain,
+		GDPRConsent:      missenaParams.GDPRConsent,
+		GDPR:             missenaParams.GDPR,
+		Placement:        missenaParams.Placement,
+		TestMode:         missenaParams.TestMode,
+	}
+	return &missenaRequest
+}
+
+func (a *adapter) makeRequest(missenaParams MissenaInternalParams, reqInfo *adapters.ExtraRequestInfo, existingRequests []*adapters.RequestData, request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+
+	url := a.endpoint + "?t=" + missenaParams.ApiKey
+	parameter := a.makeParameter(missenaParams, request)
+	body, errm := json.Marshal(parameter)
+	if errm != nil {
+		return nil, errm
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	if request.Device != nil {
+		headers.Add("User-Agent", request.Device.UA)
+	}
+
+	if request.Device != nil {
+		if request.Device.IP != "" {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		} else if request.Device.IPv6 != "" {
+			headers.Add("X-Forwarded-For", request.Device.IPv6)
+		}
+	}
+	if request.Site != nil {
+		headers.Add("Referer", request.Site.Page)
+	}
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     url,
+		Headers: headers,
+		Body:    body,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	var httpRequests []*adapters.RequestData
+	var tempErrors []error
+	gdprApplies, consentString := readGDPR(request)
+
+	var missenaInternalParams MissenaInternalParams
+
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			tempErrors = append(tempErrors, &errortypes.BadInput{
+				Message: "Error parsing bidderExt object",
+			})
+			continue
+		}
+
+		var missenaExt openrtb_ext.ExtImpMissena
+		if err := json.Unmarshal(bidderExt.Bidder, &missenaExt); err != nil {
+			tempErrors = append(tempErrors, &errortypes.BadInput{
+				Message: "Error parsing missenaExt parameters",
+			})
+			continue
+		}
+
+		missenaInternalParams.ApiKey = missenaExt.ApiKey
+		missenaInternalParams.Placement = missenaExt.Placement
+		missenaInternalParams.TestMode = missenaExt.TestMode
+	}
+
+	var finalErrors []error
+	missenaInternalParams.GDPR = gdprApplies
+	missenaInternalParams.GDPRConsent = consentString
+
+	newHttpRequest, err := a.makeRequest(missenaInternalParams, requestInfo, httpRequests, request)
+
+	if len(tempErrors) > 0 {
+		return nil, tempErrors
+	}
+
+	if err != nil {
+		finalErrors = append(finalErrors, err)
+	} else if newHttpRequest != nil {
+		httpRequests = append(httpRequests, newHttpRequest)
+	}
+
+	return httpRequests, finalErrors
```

Comment:
Do you need the `else if`? You could probably just do:
```
httpRequests = append(httpRequests, newHttpRequest)
return httpRequests, errors
```

Reply:
Done ✅